### PR TITLE
feat: Support self-defined notify webhook

### PR DESCRIPTION
**Pull request by foshantiger**
Self-defined notify webhook makes it possible to support more notify channels and to control the behavior of the messages, such as storage and archive.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -78,6 +78,7 @@ require (
 	github.com/openzipkin/zipkin-go v0.2.2 // indirect
 	github.com/parnurzeal/gorequest v0.2.15
 	github.com/pborman/uuid v1.2.0
+	github.com/pierrec/lz4 v2.0.5+incompatible // indirect
```
Comment (John-Lin) [inline go.mod]: It’s seem no need to import this package?

Comment (foshantiger) [reply go.mod]: Yep, this line add automatically to the go.mod file in my dev environment.

Comment (foshantiger) [reply go.mod]: > It’s seem no need to import this package?

Yep, this line add automatically to the go.mod file in my dev environment.

**Pull request merged**